### PR TITLE
Revert "chore: update libp2p to 0.52.1 (#14429)"

### PR DESCRIPTION
**Pull request by altonen**
This reverts commit d38d176b844aab1338ce79eb71cd6df86c97d4a0.

polkadot companion: https://github.com/paritytech/polkadot/pull/7583
cumulus companion: https://github.com/paritytech/cumulus/pull/2975

### CODE REVIEW

**Review by dmitry-markin — APPROVED** [review]
Is it required due to `V1Lazy` stuff?

**Review by vstakhov — APPROVED** [review]
I really don't like the sense of reverting updates, but it seems that 0.51 behaves more stable in terms of network connectivity (at least as we can see on Versi).

File: frame/support/Cargo.toml [hunk 1]
```diff
@@ -35,10 +35,10 @@ paste = "1.0"
 sp-state-machine = { version = "0.28.0", default-features = false, optional = true, path = "../../primitives/state-machine" }
 bitflags = "1.3"
 impl-trait-for-tuples = "0.2.2"
-smallvec = "1.11.0"
```
Comment (ggwpez) [inline frame/support/Cargo.toml:38]: Does this also have to be reverted?  
Otherwise please undo the reversions in the `frame` folder.

**Comment by altonen** [comment]
@dmitry-markin 

It's due to DHT issues we're experiencing on Versi. I'll get back to this upgrade soon enough but in the mean time we have to revert because I don't know how long it's going to take to figure out the underlying issues and there are other important tasks as well.

**Comment by rphmeier** [comment]
This should be merged, no?

As far as we know, libp2p 0.52.1 is broken and should be taken out of `master` ASAP, right?

**Comment by altonen** [comment]
> This should be merged, no?
> 
> As far as we know, libp2p 0.52.1 is broken and should be taken out of `master` ASAP, right?

yeah, just missing one last review from cumulus companion

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3393778

**Review by sandreim — APPROVED** [review]

**Review by kpp — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3397119

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3397106

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3397107

**Comment by bkchr** [comment]
bot merge force

**Comment by mxinden** [comment]
Let me know in case you need any help debugging the `v0.52.1` issues you are seeing.

- Expanded on `V1Lazy` in https://github.com/paritytech/substrate/pull/14691#issuecomment-1675946673.
- Setting [`set_mode(Mode::Server)`](https://docs.rs/libp2p-kad/latest/libp2p_kad/struct.Kademlia.html#method.set_mode) will revert to the previous Kademlia behavior, though as one would expect, disables the Kademlia client mode optimization.

**Comment by altonen** [comment]
@mxinden cheers, I'll be in contact with you when I start working the upgrade again and if I can't figure out the issues.

**Pull request merged**